TypeCobol/Compiler/Sql/CodeElements/SqlCodeElementBuilder.cs

```diff
@@ -49,19 +55,85 @@ private SelectClause CreateSelectClause(CodeElementsParser.Sql_selectClauseConte
             SyntaxProperty<SelectionModifier> selectionModifier = null;
             if (context.SQL_ALL() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All, ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.All,
+                    ParseTreeUtils.GetFirstToken(context.SQL_ALL()));
             }
             else if (context.SQL_DISTINCT() != null)
             {
-                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct, ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
+                selectionModifier = new SyntaxProperty<SelectionModifier>(SelectionModifier.Distinct,
+                    ParseTreeUtils.GetFirstToken(context.SQL_DISTINCT()));
             }
+
             if (context.star() != null)
             {
                 return new SelectClause(selectionModifier, new StarSelection());
             }
 
+            if (context.selections() != null)
+            {
+                var tableOrViewAllColumnsSelections = context.selections().selection().Select(CreateSelection)
+                    .Where(selection => selection != null).ToList();
+
+                return new SelectClause(selectionModifier, tableOrViewAllColumnsSelections);
+            }
+
+            return null;
+        }
+
+        private TableOrViewAllColumnsSelection CreateSelection(CodeElementsParser.SelectionContext context)
+        {
+            if (context.tableOrViewAllColumnsSelection() != null)
+            {
+                var tableOrViewOrCorrelationName =
+                    context.tableOrViewAllColumnsSelection().tableOrViewOrCorrelationName();
+                Token name = tableOrViewOrCorrelationName.Name as Token;
+                Token schemaName = tableOrViewOrCorrelationName.SchemaName as Token;
+                Token dbms = tableOrViewOrCorrelationName.DBMS as Token;
+                TableViewCorrelationName fullName = CreateSymbolReference(name, schemaName, dbms);
+                return new TableOrViewAllColumnsSelection(fullName);
+
+            }
+
             return null;
+        }
 
+        private TableViewCorrelationName CreateSymbolReference(Token nameToken, Token qualifierToken,
```

Comment:
In this PR we'll keep `TableViewCorrelationName` class but discard `SqlStorageArea` so to help factorize the code:
- revert to `private QualifiedSymbolReference CreateSymbolReference(Token nameToken, Token qualifierToken, Token topLevelQualifierToken)` so as to keep a generic method able to build qualified names
- wrap the result of this method into a `TableViewCorrelationName` in the `CreateSelection` method
  - the `TableViewCorrelationName` will now accept a QualifiedSymbolReference argument in its constructor instead of a `SqlStorageArea`